### DISCUSSION TITLE
Coordinate system conversion

### DISCUSSION THREAD

**Discussion by warflash** [visgl/deck.gl#7334] [Q&A] [answered]
👋
At the moment I'm struggeling with converting to and from cartesian/web mercator systems a bit and figured I'd ask for help here.

The base of my map is a tilelayer on which I want to draw a pointcloudlayer. The tiles as well as the points are game coordinates so cartesian. I wonder if there is a good way to make tile/terrain layers work well with cartesian coords - at the moment I'm using a hand crafted model matrix + coord origin + `COORDINATE_SYSTEM.CARTESIAN` to somehow brute force things in the proper position.

That works but no I have no idea how to convert the latlng values from the viewstate into cartesian coords or the other way round.

I'd appreciate any advice on if there's just a better solution to begin with skipping the whole weird transformation or there isn't - how would I go about manually converting between the layers (required for things like pan to gamecoord x/y)

Hope that makes sense, thank ya'll!

**Comment by Pessimistress** [comment]
Is there a reason you have to use the MapView?

**Reply by warflash** [reply]
Good question, so one requirement is supporting terrain as well as tile layers so it seemed like MapView would be the only choice. 
The current version can be found [here](https://bdolytics.com/map) in case that helps show what I'm trying to achieve, if needed I can also put together a small repro with the current setup :)

**Reply by Pessimistress** [reply]
TileLayer works with both geospatial and non-geospatial views. If TerrainLayer does not work with non-geospatial you can open a bug.

**Reply by charlieforward9** [reply, marked as answer]
[See/Contribute Docs](https://deck.gl/docs/developer-guide/coordinate-systems)
